store/s3.go

```diff
@@ -131,13 +139,44 @@ func (s *S3) Delete(key string) error {
 		Bucket: aws.String(s.Bucket),
 		Key:    aws.String(s.Prefix + key),
 	})
+	if err != nil {
+		log.Println("S3 Delete:", s.Prefix, key, err)
+		raven.CaptureError(err, map[string]string{"Bucket": s.Bucket, "Prefix": s.Prefix, "Key": key})
+	} else {
+		s.setkeysize(key, -1)
```

Comment:
Tell me more `-1`, looks like a magic number.

Reply:
It is. look at comment on line 33.

Reply:
What are your thoughts on a constant that names the magic?

Reply:
Maybe a constant could work? But we are not treating the field as an enumeration. If >0 it is a size, if == 0 it is unknown, if <0 it is marked as deleted.

I changed the code, and this constant would only be used here since in stat() the comparison is <0 and not == -1. (see line 163)